qemu/tests/virtio_mem_reboot.py

```diff
@@ -0,0 +1,64 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Virtio-mem reboot test
+    1) Boot guest
+    2) Check virtio-mem device
+    4) Resize virtio-mem device
+    5) Check virtio-mem device
+    6) Resize virtio-mem device
+    7) Check virtio-mem device
+    8) Reboot guest
+    9) Check virtio-mem device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    timeout = params.get_numeric("login_timeout", 240)
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    virtio_mem_model = 'virtio-mem-pci'
+    if '-mmio:' in params.get("machine_type"):
+        virtio_mem_model = 'virtio-mem-device'
+    vmem_dev = vm.devices.get_by_params({'driver': virtio_mem_model})[0]
+    device_id = vmem_dev.get_qid()
+    requested_size_vmem_test = params.get("requested-size_test_vmem0")
+    node_id = int(vmem_dev.get_param('node'))
+    req_size = vmem_dev.get_param('requested-size')
+    initial_req_size = str(int(float(normalize_data_size(req_size, 'B'))))
+
+    virtio_mem_utils.check_memory_devices(device_id, initial_req_size,
+                                          threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem(node_id, initial_req_size,
+                                            threshold, vm, test)
+    for requested_size in requested_size_vmem_test.split():
+        req_size_normalized = int(float(normalize_data_size(requested_size,
+                                  'B')))
+        vm.monitor.qom_set(device_id, "requested-size",
+                           req_size_normalized)
+        time.sleep(30)
+        virtio_mem_utils.check_memory_devices(device_id, requested_size,
+                                              threshold, vm, test)
+        virtio_mem_utils.check_numa_plugged_mem(node_id, requested_size,
+                                                threshold, vm, test)
+    vm.reboot()
+    error_context.context("Verify virtio-mem device after reboot",
+                          test.log.info)
+    virtio_mem_utils.check_memory_devices(device_id, requested_size,
+                                          threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem(node_id, requested_size,
+                                            threshold, vm, test)
+    session.close()
```

Comment:
Seems the session is not used, how about just setting `vm.wait_for_login()` in above so that we don't need to close it?